doc/src/guides/solving/solve-matrix-equation.md

```diff
@@ -0,0 +1,346 @@
+# Solve a Matrix Equation Algebraically
+
+Use SymPy to solve a matrix (linear) equation. For example, solving $
+\left[\begin{array}{cc} c & d\\1 & -e\end{array}\right] \left[\begin{array}{cc}
+x\\y\end{array}\right] = \left[\begin{array}{cc} 2\\0\end{array}\right] $ yields
+$ \left[\begin{array}{cc} x\\y\end{array}\right] = \left[\begin{array}{cc}
+\frac{2e}{ce+d}\\\frac{2}{ce+d}\end{array}\right]$.
+
+## Alternatives to Consider
+- If your matrix and constant vector contain only numbers, not symbols, for
+  example $\left[\begin{array}{cc} 1 & 2\\3 & 4\end{array}\right]
+\left[\begin{array}{cc} x\\y\end{array}\right] = \left[\begin{array}{cc}
+  2\\0\end{array}\right]$, you can use one of these other free and open-source
+  packages instead of SymPy:
+    - NumPy's {external:func}`numpy.linalg.solve`
+    - SciPy's {external:func}`scipy.linalg.solve`
+    - mpmath's
+      [lu_solve()](https://mpmath.org/doc/current/matrices.html#linear-equations)
+- Solving a matrix equation is equivalent to solving a system of linear
+  equations, so if you prefer you can
+  [](solve-system-of-equations-algebraically.md)
+- If you formulated your problem as a system of linear equations, and want to
+  convert it to matrix form, you can use {func}`~.linear_eq_to_matrix` and then
+  follow the procedures in this guide.
+
+## Solve a Matrix Equation
+
+Here is an example of solving a matrix equation with SymPy's
+{meth}`sympy.matrices.matrices.MatrixBase.solve`. We use the standard matrix
+equation formulation $Ax=b$ where
+- $A$ is the matrix representing the coefficients in the linear equations
+- $x$ is the column vector of unknowns to be solved for
+- $b$ is the column vector of constants, where each row is the value of an
+  equation
+
+```py
+>>> from sympy import init_printing
+>>> init_printing(use_unicode=True)
+```
+
+```py
+>>> from sympy import symbols
+>>> from sympy.matrices import Matrix
+>>> c, d, e = symbols("c, d, e")
+>>> A = Matrix([[c,d], [1, -e]])
+>>> A
+⎡c  d ⎤
+⎢     ⎥
+⎣1  -e⎦
+>>> b = Matrix([2, 0])
+>>> b
+⎡2⎤
+⎢ ⎥
+⎣0⎦
+>>> A.solve(b)
+⎡  2⋅e  ⎤
+⎢───────⎥
+⎢c⋅e + d⎥
+⎢       ⎥
+⎢   2   ⎥
+⎢───────⎥
+⎣c⋅e + d⎦
+```
+
+## Guidance
+
+### Matrix Usually Must Be Square
+
+The matrix $A$ usually must be square to represent a system of linear equations
+with the same number of unknowns as equations. If not, SymPy will give the error
+``ShapeError: `self` and `rhs` must have the same number of rows.``
+
+The exception to the requirement that a matrix be square comes from SymPy's use
```

Comment:
@moorepants If there's a concise explanation of what circumstances the Moore-Penrose pseudoinverse allows for a non-square matrix under, I would appreciate you sharing it. (That's beyond my math knowledge.) Thanks!